tests/Avalonia.Base.UnitTests/Collections/AvaloniaListTests.cs

```diff
@@ -148,6 +148,23 @@ public void Adding_Items_Should_Raise_CollectionChanged()
             Assert.True(raised);
         }
 
+        [Fact]
+        public void AddRange_Items_Should_Raise_Correct_CollectionChanged()
+        {
+            var target = new AvaloniaList<object>();
+
+            var eventItems = new List<object>();
+
+            target.CollectionChanged += (sender, args) =>
+            {
+                eventItems.AddRange(args.NewItems.Cast<object>());
+            };
+            
+            target.AddRange(Enumerable.Range(0,10).Select(i => new object()));
```

Comment:
If it is unclear what is happening here: If we were to so something like return random numbers instead of new objects from the `Enumerable.Select` -  `eventItems` and `target` will contain completely different numeric values .